### ISSUE TITLE
Finer grained control over escaping/unescaping entities

### ISSUE THREAD

**Issue by davidsantiago** [jhy/jsoup#275]
I'm trying to parse some HTML and be able to switch between having entities be unescaped or not. As it currently stands, I'm able to use TextNode/getWholeText() to retrieve a text node's contents such that whitespace is preserved, but entities are unescaped. Or, I can use TextNode/outerHtml() to get the node's contents such that entities are left escaped, but whitespace is normalized. As far as I can tell, there is no way to either 
- a) get the raw original HTML source of a text node 
- b) ask for the contents of a text node with the white space preserved and entities escaped 
- c) Take the contents of a text node, and manually escape/unescape it

Regarding a): It seems like the two options are TextNode/outerHtml and TextNode/getWholeText, which makes you choose between whitespace preservation and entities escaped. I've tried to get the "text" attribute out, but even though it prints escaped+whitespace-preserved when it appears in an Attributes object returned from TextNode/attributes(), there does not appear to be a way to get it back out manually. In the Attribute object, it appears to be stored unescaped, and only gets escaped again when it is printed, via the Attribute/html() function. Which would be great, except it also prints the attribute key at the same time, so there is no way to get the value "a la carte." As far as I can tell, there's no way to get the value I see when I call .toString() on an Attributes object from a TextNode, although the value I want to be getting is printed out inside the "#<Attributes ...>" output. 

Regarding b): It seems pretty clear there is no function that gets you these things with fine-grained control over this from a TextNode. As I said, I am only aware of TextNode/outerHtml() and TextNode/getWholeText(), and TextNode/text(). The first gives you escaped entities and normalized whitespace, the second gives you unescaped entities and preserved whitespace, and the last gives you unescaped entities and normalized whitespace. There's no option for escaped entities and preserved whitespace. 

Regarding c): The functions Entities/escape() and Entities/unescape() seem like great functions. I'd love to just have access to them, but they are restricted to package access. There is probably a good reason they are not publicly available, but getting this process correct is a lot of work, and Jsoup appears to contain nearly 30k of source code just devoted to the entity mappings in the property files. This is not an easy thing to rewrite when you need it, so it would be great if Jsoup could just make that functionality available. 

In summary, any of those three routes would deliver the functionality I need. I think a) and c) are both things that would improve the library's utility, especially if they are both implemented. They are both about simply making functionality already present the library available to users of the library. Adding a fourth function to cover the last case in option b) woud also solve my problem, but seems less generally useful, and does may not map as well to HTML DOM functions. 

If you think this would be worth implementing, let me know how you'd like to see it done and I can send a patch your way.

**Comment by jhy** [comment]
(Closing out old, dormant bugs. If you are still impacted by this, please reopen & vote.)
